core/processor/abci.go

```diff
@@ -66,6 +67,7 @@ var (
 	ErrAssetProposalDisabled                         = errors.New("asset proposal disabled")
 	ErrAwaitingCheckpointRestore                     = errors.New("transactions not allowed while waiting for checkpoint restore")
 	ErrOracleNoSubscribers                           = errors.New("there are no subscribes to the oracle data")
+	ErrTransactionExpired                            = errors.New("transaction expired")
```

Comment:
I think a static error makes more sense here than `fmt.Errorf`. This error is used in one specific case, so when seeing this error, it's clear what happened.

Reply:
How about a compromise. "The transaction expired because it was not included(?) in the defined(?) block"

Reply:
I think Candida's suggestion is the right one indeed. What matters to me is the clarity of the error message. We need to make user facing error message very clear, otherwise it tends to create a lot of support ticket for explanations.